### PR TITLE
Alternative approach to handling timeouts for engines

### PR DESCRIPTION
**Pull request by noahd1**
Alternative to own handling of timeouts using stdlib `Timeout::timeout`.

/cc @codeclimate/review @brynary

### CODE REVIEW

File: lib/cc/analyzer/engine.rb [hunk 1]
```diff
@@ -35,32 +34,30 @@ def run(stdout_io, stderr_io = StringIO.new)
...
-            Analyzer.statsd.increment("cli.engines.names.#{name}.result.error.timeout")
-            raise EngineTimeout, "engine #{name} ran past #{TIMEOUT} seconds and was killed"
+        begin
+          status = nil
+          Timeout::timeout(TIMEOUT) do
+            pid, status = Process.waitpid2(pid)
           end
+        rescue Timeout::Error
+          run_command("docker kill #{container_name} || true")
+          timed_out = true
```
Comment (pbrisbin) [inline lib/cc/analyzer/engine.rb]: Can we get rid of `timed_out` and do whatever happens on `timed_out = true` in the `rescue`? I think the only change in behavior is that `"cli.engines.finished"` wouldn't be ticked in the case of a timeout. If that's important, I'd just duplicate the `increment` call: once in the happy path and again in the timeout rescue.

Also, the `begin`/`rescue` is telling me to extract a `#wait_for_container` method.

Comment (noahd1) [inline lib/cc/analyzer/engine.rb]: All good suggestions. Just waiting for someone (@brynary) to tell us why this is a bad idea first.

Comment (brynary) [inline lib/cc/analyzer/engine.rb]: In the timeout case, the process does not get cleaned up.

Comment (pbrisbin) [inline lib/cc/analyzer/engine.rb]: Calling `waitpid` again after sending the kill should reap the process.

**Pull request closed without merging**